### PR TITLE
save robot added to cart to backend

### PR DESCRIPTION
**Pull request by taherkheli**
@Jaxwood please take your opportunity to moan :p

### CODE REVIEW

File: client/src/store/actions.ts [hunk 1]
```diff
@@ -16,6 +18,16 @@ export const actions: ActionTree<State, State> = {
       console.log(error);
     }
   },
+  async [ActionTypes.AddRobotToServer]({ commit, state }, robot: Cart) {
```
Comment (Jaxwood) [inline client/src/store/actions.ts]: @taherkheli why call it `robot` if it is a `Cart`?

File: client/src/store/actions.ts [hunk 2]
```diff
@@ -1,10 +1,12 @@
 import { ActionTree } from 'vuex';
 import axios from 'axios';
+import Cart from '@/data/Cart';
 import MutationTypes from './mutations';
 import { State } from './state';
 
 enum ActionTypes {
-  GetParts = 'GET_PARTS'
+  GetParts = 'GET_PARTS',
+  AddRobotToServer = 'ADD_ROBOT_TO_SERVER'
```
Comment (Jaxwood) [inline client/src/store/actions.ts]: @taherkheli isn't it better to call it `SAVE_CART`?

**Review by Jaxwood — APPROVED** [review]

File: client/src/store/actions.ts (hunk 1 above)
Comment (taherkheli) [reply client/src/store/actions.ts:21]: done

Comment (taherkheli) [reply client/src/store/actions.ts]: @Jaxwood  its actually a `robot` with an extra field `cost` appended to it. `Cart` is not a `cart `but an enhanced `robot `with price. Cart extends Robot Interface. Chief Architect JLo named it so, i did not dare changing it ... let it be I have had enough of robots :)

**Pull request merged**